src/web/src/app/space/add/component.ts

```diff
@@ -0,0 +1,124 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { Observable } from 'rxjs';
+import { startWith, map, debounceTime, distinctUntilChanged, switchMap, tap } from 'rxjs/operators';
+import { MatOption } from '@angular/material/core';
+import { Router, NavigationExtras } from '@angular/router';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+
+import { SpaceService } from '../service';
+import { Space } from '../model';
+
+@Component({
+	selector: 'app-space-add',
+	templateUrl: './component.html',
+	styleUrls: ['./component.css', '../common.css'],
+})
+export class SpaceAddComponent implements OnInit {
+	createForm: FormGroup;
+	spaceAutocomplete = new FormControl('');
+
+	spaces$!: Observable<Space[]>;
+	selectedSpace: Space;
+	keyWasChanged: boolean = false;
+	isSpaces: boolean;
+	firstGetList: boolean = true;
+	isGettingSpaces: boolean = false;
+
+	constructor(
+		private router: Router,
+		private fb: FormBuilder,
+		private space: SpaceService,
+	) {
+		this._createForm();
+	}
+
+	ngOnInit() {
+		this.spaces$ = this.spaceAutocomplete.valueChanges.pipe(
+			tap(_ => { this.isGettingSpaces = true; }),
+			startWith(''),
+			debounceTime(300), // Optional: debounce input changes to avoid excessive requests
+			distinctUntilChanged(), // Optional: ensure distinct values before making requests
+			switchMap(value => {
+				if (!value)
+					this.selectedSpace = null;
+				return this.space.getList(10, 0, value || '').pipe(
```

Comment:
Тут мы должны получить список доступных для вступления, а не список тех в которых уже состоит пользователь.
```suggestion
				return this.space.getAvailableList(10, 0, value || '').pipe(
```

Reply:
Готово.